src/vsphere_cpi/lib/cloud/vsphere/vm_config.rb

```diff
@@ -56,10 +84,60 @@ def storage_policy_name
       @manifest_params[:storage_policy]
     end
 
+    def human_readable_name_enabled?
+      @manifest_params[:enable_human_readable_name]
+    end
+
+    def human_readable_name_info
+      @manifest_params.fetch(:human_readable_name_info, {})
+    end
+
     def networks_spec
       @manifest_params[:networks_spec] || {}
     end
 
+    def generate_human_readable_name(instance_name, deployment_name)
```

Comment:
This code block looks really complicated for what it's trying to do (especially as it's not going to handle Unicode). Can we simplify this?

Reply:
Simplified this by eliminating unicode part